### PR TITLE
web: improve `<Button />` type

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

To allow doing `<Button as={Whatever} whateverProp={true} />` use `ForwardReferenceComponent`.

### CODE REVIEW

File: client/wildcard/src/components/Button/story/Button.story.tsx [hunk 1]
```diff
@@ -63,7 +63,6 @@ export const AllButtons = () => (
         <Button
             variant="secondary"
             as="a"
-            // @ts-expect-error (providing `as` changes possible props)
```
Comment (vovakulikov) [inline client/wildcard/src/components/Button/story/Button.story.tsx:66]: 👍

**Review by vovakulikov — APPROVED** [review]
LGTM

**Comment by vovakulikov** [comment]
Just realized that this is still in draft status. So far so good :)

File: client/web/src/batches/RepoBatchChangesButton.tsx [hunk 2]
```diff
@@ -31,7 +32,10 @@ export const RepoBatchChangesButton: FC<RepoBatchChangesButtonProps> = ({
     const { open, merged } = stats.changesetsStats
 
     return (
-        <Link className={className} to={`/${encodeURIPathComponent(repoName)}/-/batch-changes`}>
+        <Link
+            className={classNames('btn btn-outline-secondary', className)}
```
[the remark this answers is not in the document]
Comment (valerybugakov) [reply client/web/src/batches/RepoBatchChangesButton.tsx:36]: Right, it will be handled in the PR created by codemod.

**Pull request merged**